backend/whatsapp.py

```diff
@@ -239,17 +257,33 @@ def generateQRCode(self, callback=None):
         self.activeWs.send(message);
 
     def restoreSession(self, callback=None):
+        with open("session.json","r") as f:
+            session_file = f.read()
+        session = json.loads(session_file)
+        self.connInfo["clientToken"] = session['clientToken']
+        self.connInfo["serverToken"] = session['serverToken']
+        self.loginInfo["clientId"] = session['clientId']
+        self.loginInfo["key"]["macKey"] = session['macKey'].encode("latin_1")
```

Comment:
❓ does the WhatsApp server run in `latin_1` or in UTF8?

Reply:
i remember that i couldn't **json.dumps** `mackey` and `enckey` as string and get UTF-8 decoder error!
so decoded to latin_1 to save it as string and in restore i encoded to latin_1 again.